beacon-chain/blockchain/tracked_proposer.go

```diff
@@ -0,0 +1,23 @@
+package blockchain
+
+import (
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/cache"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+)
+
+// trackedProposer returns whether the beacon node was informed, via the
+// validators/prepare_proposer endpoint, of the proposer at the given slot.
+// It only returns true if the tracked proposer is present and active.
+func (s *Service) trackedProposer(st state.ReadOnlyBeaconState, slot primitives.Slot) (cache.TrackedValidator, bool) {
+	id, err := helpers.BeaconProposerIndexAtSlot(s.ctx, st, slot)
+	if err != nil {
+		return cache.TrackedValidator{}, false
+	}
```

Comment:
Isn't this redundant? Is this meant to be a defensive check in case the validator client registered a validator that's not really a proposer?

Reply:
This function takes a slot and a state, so we first need to get who is proposing in that slot.